hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/AbstractRecordStore.java

```diff
@@ -120,7 +121,12 @@ protected void addJsonMetadataMutationObserver() {
 
     @Override
     public InMemoryFormat getInMemoryFormat() {
-        return inMemoryFormat;
+        return getMapContainer().getMapConfig().getInMemoryFormat();
```

Comment:
Seems like I was thinking that in-memory format can be changed in runtime (and also make `getInMemoryFormat` consistent with `getEvictionPolicy`), if not I will revert back to reading the field

Reply:
I don't think we provide this feature (maybe there is some hacky way to change the in-memory format that I don't know of) but even if someone is able to change the in-memory format, it's definitely not a feature we support and I expect bugs to be popping up since I think a lot of code relies on the fact that the format is set during map container/record store instantiation time and that it doesn't change. We even have different record store implementations for different in-memory formats (I think one for on-heap, another for NATIVE).

Reply:
reverted back to the field